python/lsst/dax/obscore/obscore_exporter.py

```diff
@@ -331,40 +464,89 @@ def _make_schema(self, table_spec: ddl.TableSpec) -> Schema:
 
         return pyarrow.schema(schema)
 
-    def _make_record_batches(self, batch_size: int = 10_000) -> Iterator[RecordBatch]:
+    def _make_record_batches(
+        self, batch_size: int = 10_000, limit: int | None = None
+    ) -> Iterator[RecordBatch]:
         """Generate batches of records to save to a file."""
         batch = _BatchCollector(self.schema)
 
+        # Reset overflow flag.
+        self.overflow = False
+
         collections: Any = self.config.collections
         if not collections:
-            collections = ...
+            raise ValueError("No collections specified. Querying all collections is not allowed.")
+
+        if limit is None:
+            unlimited = True
+        elif limit == 0:
+            # Return immediately.
+            return
+        else:
+            unlimited = False
+            # Always ask for one extra to allow the overflow detection.
+            query_limit = abs(limit) + 1
```

Comment:
Maybe `unlimited` -> `is_unlimited`. I'll have a go at tweaking things. `unlimited` is separate because `limit == 0` means something other than unlimited.

Reply:
Also I was using a different variable name because `limit` is `int | None` but `query_limit` is `int`. `limit` can be negative but `query_limit` must be positive. It seemed simpler to keep them distinct. I can drop `unlimited` and replace it with a check for `limit is None` although `is_unlimited` in the loop seems clearer.

Reply:
Ok. Maybe I was too hasty. `limit` being negative is not actually allowed -- I was thinking of the butler command lines -- and we always ask for limit+1.